src/include/network/lan/address.rb

```diff
@@ -728,36 +704,96 @@ def HandleSlave(_key, event)
     # @param [String] key	id of the widget
     # @param [String] key id of the widget
     def StoreSlave(_key, _event)
-      configured_slaves = Ops.get_list(@settings, "SLAVES", [])
+      configured_slaves = @settings["SLAVES"] || []
 
-      Ops.set(
-        @settings,
-        "SLAVES",
-        Convert.convert(
-          UI.QueryWidget(:msbox_items, :SelectedItems),
-          from: "any",
-          to:   "list <string>"
-        )
-      )
-      Ops.set(@settings, "BONDOPTION", UI.QueryWidget(Id("BONDOPTION"), :Value))
+      @settings["SLAVES"] = UI.QueryWidget(:msbox_items, :SelectedItems) || []
+
+      @settings["BONDOPTION"] = UI.QueryWidget(Id("BONDOPTION"), :Value).to_s
 
-      LanItems.bond_slaves = Ops.get_list(@settings, "SLAVES", [])
-      LanItems.bond_option = Ops.get_string(@settings, "BONDOPTION", "")
+      LanItems.bond_slaves = @settings["SLAVES"]
+      LanItems.bond_option = @settings["BONDOPTION"]
 
       # create list of "unconfigured" slaves
-      new_slaves = Builtins.filter(Ops.get_list(@settings, "SLAVES", [])) do |slave|
-        !Builtins.contains(configured_slaves, slave)
+      new_slaves = @settings["SLAVES"].select do |slave|
+        !configured_slaves.include? slave
       end
 
-      Lan.bond_autoconf_slaves = Convert.convert(
-        Builtins.toset(Builtins.merge(Lan.bond_autoconf_slaves, new_slaves)),
-        from: "list",
-        to:   "list <string>"
-      )
+      Lan.bond_autoconf_slaves = (Lan.bond_autoconf_slaves + new_slaves).uniq.sort
 
       nil
     end
 
+    # Validates created bonding. Currently just prevent the user to create a
+    # bond with more than one interface sharing the same physical port id
+    #
+    # @param [String] key the widget being validated
+    # @param [Hash] event the event being handled
+    # @return true if valid or user decision if not
+    def ValidateBond(_key, _event)
+      items = UI.QueryWidget(:msbox_items, :SelectedItems) || []
+      physical_port_ids = {}
+
+      items.each do |i|
+        if has_physical_port_id?(i)
+          p = physical_port_ids[physical_port_id(i)] ||= []
+          p << i
+        end
+      end
+
+      physical_port_ids.select! { |_k, v| v.size > 1 }
```

Comment:
But this check is for having more than 1 element. If there is only one interface per physical port id then it is valid.